src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

```diff
@@ -23,7 +23,7 @@ public DirectoryInfo(String path)
         }
 
         [System.Security.SecuritySafeCritical]
-        internal DirectoryInfo(String fullPath, IFileSystemObject fileSystemObject) : base(fileSystemObject)
+        internal DirectoryInfo(String fullPath, bool ignoreThis)
```

Comment:
> The ignored parameter is unfortunate.

I agree. I just went with the ignored parameter as this is what was done [previously](https://github.com/dotnet/coreclr/blob/4cf8a6b082d9bb1789facd996d8265d3908757b2/src/mscorlib/src/System/IO/FileInfo.cs#L132). I'll hold off on improving this until I hear back on the overall direction of the proposal.